tests/js/session-tests.js

```diff
@@ -527,6 +527,24 @@ module.exports = {
       .then(() => TestCase.assertTrue(progressCalled));
   },
 
+  // test that it is possible to register a custom logging function on a
+  // Sync session, and that it is invoked when a session is established
+  async testCustomSessionLogging() {
+    const partition = Utils.genPartition();
+    let credentials = Realm.Credentials.anonymous();
+    let app = new Realm.App(appConfig);
+    let user = await app.logIn(credentials);
+    let config = getSyncConfiguration(user, partition);
+
+    let loggerInvoked = false;
+    Realm.App.Sync.setLogLevel(app, "info"); // "all", "trace", "debug", "detail", "info", "warn", "error", "fatal", "off"
+    Realm.App.Sync.setLogger(app, (level, message) => (loggerInvoked = true));
```

Comment:
I guess it might not hurt to test that `level` and `message` are passed through correctly?

```suggestion
    let loggedLevel, loggedMessage;
    Realm.App.Sync.setLogLevel(app, "info"); // "all", "trace", "debug", "detail", "info", "warn", "error", "fatal", "off"
    Realm.App.Sync.setLogger(app, (level, message) => {
      if (!loggedLevel) loggedLevel = level;
      if (!loggedMessage) loggedMessage = message;
    });
    
    // ...
    
    TestCase.assertEqual(loggedLevel, 123);
    TestCase.assertEqual(loggedMessage, "whatever the first message is");
```

Or is the first message not predictable?

Reply:
It is as long as the sync subsystem doesn't change what it returns, or the sync config structure doesn't change.  It'd be weird if the test failed because of that, but I'll see if I can parse out from the lower levels how the message is constructed.

Reply:
I wouldn't worry too much if it's awkward :)